force-app/tests/Email Recipes/InboundEmailHandlerRecipes_tests.cls

```diff
@@ -0,0 +1,140 @@
+            true
+        );
+        Contact testContact = (Contact) TestFactory.createSObject(
+            new Contact(
+                email = FROM_EMAIL,
+                accountId = testAccount.Id,
+                Lastname = 'Poorman'
+            ),
+            true
+        );
+    }
+
+    private static Messaging.inboundEmail.BinaryAttachment createAttachment(
+        String fileName,
+        Blob fileBody
+    ) {
+        Messaging.inboundEmail.BinaryAttachment attatchment = new Messaging.inboundEmail.BinaryAttachment();
+        attatchment.filename = fileName;
+        attatchment.body = fileBody;
+        return attatchment;
+    }
+
+    private static Messaging.InboundEmail createEmail() {
+        Messaging.InboundEmail email = new Messaging.InboundEmail();
+        email.fromName = 'Kevin Poorman';
+        email.fromAddress = FROM_EMAIL;
+        email.toAddresses = new List<String>{ 'invalid@invalid.com.invalid' };
+        email.subject = 'Test Email';
+        email.plainTextBody = 'Welcome CodeLive Viewers';
+        email.binaryAttachments = new List<Messaging.InboundEmail.BinaryAttachment>{
+            createAttachment(
+                'helloWorld.txt',
+                Blob.valueOf('Hello CodeLive Viewers')
+            ),
+            createAttachment(
+                'helloWorld2.txt',
+                Blob.valueOf('Hello CodeLive Viewers two')
+            )
+        };
+        return email;
+    }
+
+    private static Messaging.InboundEnvelope createEnvelope() {
+        Messaging.InboundEnvelope envelope = new Messaging.InboundEnvelope();
+        envelope.fromAddress = FROM_EMAIL;
+        return envelope;
+    }
+
+    @isTest
+    static void handleInboundEmailNegativeNoContactFound() {
+        Messaging.InboundEnvelope envelope = createEnvelope();
+        Messaging.InboundEmail email = new Messaging.InboundEmail();
+        email.fromAddress = FROM_EMAIL;
+
+        delete [SELECT id FROM Contact];
+        Test.startTest();
+        Messaging.InboundEmailResult result = new InboundEmailHandlerRecipes()
+            .handleInboundEmail(email, envelope);
+        Test.stopTest();
+
+        System.assertEquals(
+            false,
+            result.success,
+            'Expected to receive a false success property.'
+        );
+
+        System.assert(
+            result.message.containsIgnoreCase('Unable to create new contact'),
+            'Expected to recieve a InboudEmailHandlerRecipesException with the given message'
+        );
+    }
+
+    @isTest
+    static void handleInboundEmailNegativeNewContactCreated() {
+        Messaging.InboundEnvelope envelope = createEnvelope();
+        Messaging.InboundEmail email = createEmail();
+
+        delete [SELECT id FROM Contact];
+
+        Test.startTest();
+        Messaging.InboundEmailResult result = new InboundEmailHandlerRecipes()
+            .handleInboundEmail(email, envelope);
+        Test.stopTest();
+
+        System.assertEquals(
```

Comment:
This test isn't sufficient: you're testing for the reuse of an existing contact but shouldn't you also test for the creation of a new contact with the right info?
This would be a good fit for `integrationTestPositive`.

Reply:
good catch. Additional assert created.